aiida_lammps/workflows/base.py

```diff
@@ -0,0 +1,274 @@
+"""Workchain to run a LAMMPS calculation with automated error handling and restarts."""
+import re
+
+from aiida import orm
+from aiida.common import AttributeDict, InputValidationError
+from aiida.engine import (
+    BaseRestartWorkChain,
+    ProcessHandlerReport,
+    process_handler,
+    while_,
+)
+import numpy as np
+
+from aiida_lammps.calculations.base import LammpsBaseCalculation
+from aiida_lammps.utils import RestartTypes
+
+
+class LammpsBaseWorkChain(BaseRestartWorkChain):
+    """Base workchain for calculations using LAMMPS"""
+
+    _process_class = LammpsBaseCalculation
+
+    @classmethod
+    def define(cls, spec):
+        """Define the process specification."""
+        # yapf: disable
+        super().define(spec)
+        spec.expose_inputs(LammpsBaseCalculation)
+        spec.input(
+            "max_iterations",
+            valid_type=orm.Int,
+            default=lambda: orm.Int(5),
+            help="Maximum number of restarts",
+        )
+        spec.input(
+            "clean_workdir",
+            valid_type=orm.Bool,
+            required=False,
+            default= lambda: orm.Bool(False),
+            help="""
+            Whether to clean the workdir of the calculations or not,
+            the default is not clean.
+            """,
+        )
+        spec.input(
+            "verbosity",
```

Comment:
I don't think I see this used anywhere. Anyway, I think it might be better to just use the logger with the various levels that already exist. You can always use `self.logger.warning` or `self.logger.info` to log messages at various levels and then the user can control the level with the global `logging.aiida_loglevel` flags.

Reply:
Nice, I did not know about this, in the aiida-vasp there is this flag that can be used to decide whether some messages should be printed or not via the `self.report`.

Does the logger work in a similar way?

I know it does nothing right now but I wanted to start planning for it, but if it can be done via the logger it is much better as it does not generate more nodes that one does not  really need

Reply:
`self.report` simply forwards to `self.logger.report`. It simply logs a message on the `REPORT` level, which is a level AiiDA defines in between `INFO` and `WARNING`. If you want to log verbose information that is not shown by default, I recommend you simply use `self.logger.info(message)`. This will not show up by default, but the user can set `verdi config set logging.aiida_loglevel INFO` and then they will show. This controls logging for the entire application. Think this is better then each workflow adding custom solutions.

Reply:
Fantastic, I was not aware of it being perfectly honest. Will also forward this to `aiida-vasp`, I think that this is nice, since if we do not need to store those nodes it is much better.